File: 06_quadratic_equation/quadratic_equation.js
//quadratic equation ax2 + bx + c = 0
var a = 2;
var b = 0;
var c = -1;

var D = b * b - 4 * a * c;
if (D < 0) {
  console.log("No solution in R");
} else if (D > 0) {
  var x1 = Math.round(((-b + Math.sqrt(D)) / (2 * a)) * 100) / 100;
  var x2 = Math.round(((-b - Math.sqrt(D)) / (2 * a)) * 100) / 100;
  console.log("Two solution: x1 = " + x1 + ", x2 = " + x2);
} else {
  console.log("One solution: x = " + -b / (2 * a));
}

File: 06_quadratic_equation/quadratic_equation.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main()
{
    //quadratic equation ax2 + bx + c = 0
    double a = 2;
    double b = 0;
    double c = 1;
    double x1, x2;

    double D = (b * b) - (4 * a * c);
    if (D < 0)
    {
        puts("No solution in R");
    }
    else if (D > 0)
    {
        x1 = (-b + sqrt(D)) / (2 * a);
        x2 = (-b - sqrt(D)) / (2 * a);
        printf("Two solution: x1 = %.2f, x2 = %.2f\n", x1, x2);
    }
    else
    {
        x1 = (-b) / (2 * a);
        printf("One solution: x = %.2f\n", x1);
    }
    return 0;
}